cmd/av/stack_tree.go

```diff
@@ -71,14 +71,89 @@ func printStackTree(repo *git.Repo, branches map[string]meta.Branch, currentBran
 		fmt.Printf("%s<ERROR: unknown branch: %s>\n", indent, root)
 		return
 	}
+
+	branchInfo, err := getBranchInfo(repo, branch)
+	if err != nil {
+		fmt.Printf("<ERROR: cannot get branch info: %v>\n", err)
+		return
+	}
+
 	if currentBranch == branch.Name {
 		_, _ = fmt.Print(
-			indent, colors.Success("* "), colors.Success(branch.Name), "\n",
+			indent, colors.Success("* "), colors.Success(branch.Name), " ", colors.Faint(branchInfo), "\n",
 		)
 	} else {
-		_, _ = fmt.Printf("%s%s\n", indent, branch.Name)
+		_, _ = fmt.Print(indent, branch.Name, " ", colors.Faint(branchInfo), "\n")
 	}
 	for _, next := range branch.Children {
 		printStackTree(repo, branches, currentBranch, next, depth+1)
 	}
 }
+
+func getBranchInfo(repo *git.Repo, branch meta.Branch) (string, error) {
+	var branchInfo string
+
+	parentStatus, err := getParentStatus(repo, branch)
+	if err != nil {
+		fmt.Printf("<ERROR: could not get status of parent branch: %v>\n", err)
```

Comment:
No. Do this in the caller side.

In general, if a method looks like side-effect free (like this method), it should be side effect free.